public/app/features/explore/spec/helper/assert.ts

```diff
@@ -1,7 +1,32 @@
-import { screen } from '@testing-library/react';
+import { waitFor } from '@testing-library/react';
+import { ExploreId } from '../../../../types';
+import { withinExplore } from './setup';
 
-export const assertQueryHistoryExists = (query: string) => {
-  expect(screen.getByText('1 queries')).toBeInTheDocument();
-  const queryItem = screen.getByLabelText('Query text');
+export const assertQueryHistoryExists = (query: string, exploreId: ExploreId = ExploreId.left) => {
+  const selector = withinExplore(exploreId);
+
+  expect(selector.getByText('1 queries')).toBeInTheDocument();
+  const queryItem = selector.getByLabelText('Query text');
   expect(queryItem).toHaveTextContent(query);
 };
+
+export const assertQueryHistory = async (expectedQueryTexts: string[], exploreId: ExploreId = ExploreId.left) => {
+  const selector = withinExplore(exploreId);
+  await waitFor(() => {
+    expect(selector.getByText(`${expectedQueryTexts.length} queries`)).toBeInTheDocument();
+    const queryTexts = selector.getAllByLabelText('Query text');
+    expectedQueryTexts.forEach((expectedQueryText, queryIndex) => {
+      expect(queryTexts[queryIndex]).toHaveTextContent(expectedQueryText);
+    });
+  });
+};
+
+export const assertQueryHistoryIsStarred = async (expectedStars: boolean[], exploreId: ExploreId = ExploreId.left) => {
+  const selector = withinExplore(exploreId);
+  const starButtons = selector.getAllByRole('button', { name: /Star query|Unstar query/ });
+  await waitFor(() =>
+    expectedStars.forEach((starred, queryIndex) => {
+      expect(starButtons[queryIndex].getAttribute('title')).toBe(starred ? 'Unstar query' : 'Star query');
```

Comment:
i think we can go a step further and test this better with:

```suggestion
      expect(starButtons[queryIndex]).toHaveAccessibleName(starred ? 'Unstar query' : 'Star query');
```

Reply:
Nice! Added in 38868d055943b98e9d5d1aaf5ab0b5129c63f16d